### ISSUE TITLE
Test with 5.4 kernels

### ISSUE THREAD

**Issue by shawaj** [PiSupply/Media-Center-HAT#11]
As per this issue...

https://github.com/PiSupply/PaPiRus/issues/215

**Comment by ryanteck** [comment]
It didn't work and needs fixing.

**Comment by ryanteck** [comment]
Duplicate of #10 irc

**Comment by shawaj** [comment]
For reference https://github.com/raspberrypi/linux/commit/b7d685c0b1bd1b98af0e9c1f5d43769982bdbfb2#diff-dc041e47200262dfab1ff9f9e0ef64c7

**Comment by tvoverbeek** [comment]
Just tested the changes Phil made in https://github.com/raspberrypi/linux/commit/b7d685c0b1bd1b98af0e9c1f5d43769982bdbfb2#diff-dc041e47200262dfab1ff9f9e0ef64c7.
They make the screen work again. With the current overlay I only get a black screen with back-light off (The point-shoot cam still takes pictures).
The changes are not yet in the normal repository.

For now to disable the current HAT overlay add the line `dtoverlay=` before any dtparam and dtoverlay lines in /boot/config.txt. This prevents the HAT overlay from loading (No need for soldering the address jumper for address 0x51).
For now you will have to compile the changed overlay your self, copy it to /boot/overlays and put a dtoverlay line in /boot/config.txt
For the future we should use the idea Phil suggested to put only the overlay name in the HAT, so it will always use the current one in /boot/overlays.

**Comment by shawaj** [comment]
@tvoverbeek yes I agree, will move to the overlay name method for all production going forward. Need to perform some tests with our production eeprom programmers but relatively trivial to fix if there's an issue anyway

**Comment by ryanteck** [comment]
What did we need to do to get this sorted then? Update the script to modify config?

**Comment by tvoverbeek** [comment]
@ryanteck 
For Media Center Hats with the devicetree fragment in the EEPROM do the following:
- In /boot/config.txt before any dtoverlay lines put the line `dtoverlay=`.
  This prevents the loading of the HAT device tree fragment (which does not work due to changes with SPI in the 5.4 kernel)..
- Further down add the line `dtoverlay=media-center`. This will load the dtoverlay from /boot/overlays.
  The current one is fixed for the SPI communication with the media-center hat.
This works for me.

**Comment by shawaj** [comment]
@ryanteck shall we edit the install script to handle this?
